### ISSUE TITLE
Unused variables - false positive for fields set by static method

### ISSUE THREAD

**Issue by maelstraggiotti** [Riverside-Software/sonar-openedge#331]
A local variable is reported as unused if its assigned to the return value of a class static method.
```
DEFINE VARIABLE instance AS StaticStub NO-UNDO.
instance   = StaticStub:GetInstance().
```
```
CLASS StaticStub: 

    METHOD PUBLIC STATIC StaticStub GetInstance():
        RETURN NEW StaticStub().
    END.
END CLASS.
```

The _instance_ field is getting reported.

**Comment by gquerret** [comment]
Which rule is triggered ? I've tried to setup a simple use case, and issue for "Variable assigned but not read" was raised, but not for "Unused variable".

**Comment by maelstraggiotti** [comment]
That's correct, VariableAssignedButNotRead is raised. And it's relevant, the variable was indeed not read, this issue ain't one. 

Sorry for the inconvenience.
